### PR TITLE
fix oAuth bearer and set openshift container for openshift deployment

### PR DESCRIPTION
**Pull request by zbendhiba**
fixes #2872

- Uses DevServices for dev mode, instead of docker-compose
- Adds the oAuth client for SASL Oauth Bearer
- Adds secrets for Kubernetes and Openshift deployments

### CODE REVIEW

File: kafka/README.adoc [hunk 1]
```diff
@@ -9,27 +9,16 @@ and other general information.
...
-- The section Kafka instance without Authentication if no Authentication required.
-- The section Kafka instance with SASL Plain is using SASL.
-- The section Kafka instance with SASL Oauth Bearer if using Oauth Bearer.
+For simplicity, this example uses Quarkus DevServices in dev mode. It will automatically start Strimzi container for your tests, and you don’t even need to configure anything: the container will be automatically wired to the Quarkus dev mode configuration.
 
 == Start in Development mode
 
 Run the application in development mode.
 
+TIP: If you want to use another running instance, in dev mode. Uncomment the corresponding Kafka configuration section in `src/main/resources/application.properties` and change `%prod` profile by `%dev`.
```
Comment (ppalaga) [inline kafka/README.adoc:20]:
```suggestion
TIP: If you want to use another running instance, in dev mode. Uncomment the corresponding Kafka configuration section in `src/main/resources/application.properties` and change `%prod` profile to `%dev`.
```

File: kafka/README.adoc [hunk 2]
```diff
@@ -64,13 +53,48 @@ $ curl -X GET http://0.0.0.0:8080/example
 ----
 
 
-=== Package and run the application
+=== Configure Kafka client, Package and run the application
```
Comment (ppalaga) [inline kafka/README.adoc:56]:
```suggestion
=== Configure Kafka client, package and run the application
```

File: kafka/README.adoc [hunk 3]
```diff
@@ -64,13 +53,48 @@ $ curl -X GET http://0.0.0.0:8080/example
...
+Once you are done with developing you may want to configure your kafka client, package and run the application.
 
 TIP: Find more details about the JVM mode and Native mode in the Package and run section of
 https://camel.apache.org/camel-quarkus/latest/first-steps.html#_package_and_run_the_application[Camel Quarkus User guide]
 
+==== Configure kafka client
+Uncomment the corresponding commented section in `src/main/resources/application.properties`.
+
+- The section Kafka instance without Authentication if no Authentication required.
+- The section Kafka instance with SASL Plain is using SASL.
```
Comment (ppalaga) [inline kafka/README.adoc:67]:
```suggestion
- The section Kafka instance with SASL Plain if using SASL.
```

File: kafka/README.adoc [hunk 4]
```diff
@@ -96,16 +120,36 @@ $ ./target/*-runner
...
+    </dependency>
+</dependencies>
 ----
 
 If you are using a local development cluster like Kind or k3s, you can use host the container image on your local host. Or, with minikube, use the Docker daemon from the cluster virtual machine `eval $(minikube docker-env)`. Otherwise, you'll need to push the image to a registry of your choosing.
 
-TIP: You can build & deploy in one single step by doing `mvn clean package -DskipTests -Dquarkus.kubernetes.deploy=true`
+TIP: You can build & deploy in one single step by doing `mvn clean package -DskipTests -Dkubernetes -Dquarkus.kubernetes.deploy=true`
 
 Check pods are running.
```
Comment (ppalaga) [inline kafka/README.adoc:154]:
```suggestion
Check that the pods are running.
```

File: kafka/pom.xml [hunk 5]
```diff
@@ -46,6 +46,7 @@
         <maven-surefire-plugin.version>2.22.2</maven-surefire-plugin.version>
         <mycila-license.version>3.0</mycila-license.version>
         <strimzi.testcontainers.version>0.20.1</strimzi.testcontainers.version>
+        <kafka-oauth-client.version>0.7.2</kafka-oauth-client.version>
```
Comment (ppalaga) [inline kafka/pom.xml:49]: I still wonder whether we should ask Quarkus to manage `io.strimzi:kafka-oauth-client`

**Review by ppalaga — APPROVED** [review]
Some minor comments and typos inline.

File: kafka/pom.xml (hunk 5 above)
Comment (zbendhiba) [reply kafka/pom.xml:49]: +1

**Comment by zbendhiba** [comment]
I'd delete that oauth2 profile. I don't like it. I'll do it and check why it's failing on monday

**Comment by zbendhiba** [comment]
I've pushed another commit that : 

- Deletes the oauth2 profile. I've just put the dependency, and it would be used if oauth2 
- Deletes the creation of Strimzi Test Resource. It should be managed by Quarkus
- Fixes the native test

**Review by ppalaga — APPROVED** [review]

**Pull request merged**